RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -291,32 +641,717 @@ class DeepTauId : public deep_tau::DeepTauBase {
     {
         return DeepTauBase::globalEndJob(cache_);
     }
+private:
+    static constexpr float pi = boost::math::constants::pi<float>();
```

Comment:
@perrotta After looking through the cmssw code, I see that there is no homogeneous definition of pi. In most of the places it used an old C style definition `M_PI`, in some other places it is defined as `TMath::Pi()`, or `2.0 * acos(0.)`, or, in some places, it is even hardcoded its numeric value, etc.
I personally prefer the constant defined in boost because it allows having an explicit control of the required precision. Since cmssw already strongly depends on boost libraries, I don't think that using `boost::math::constants` adds an additional dependency.
But if you prefer that pi is defined as `M_PI`, I can change it. Please, let me know.

Reply:
Well, I won't make a crusade on it, but M_PI is part of the system in CMSSW, and it looks to me as the most convenient one to be used, while avoiding extra dependencies

Reply:
@mbluj I agree with @perrotta , recently @cvuosalo has setup https://github.com/cms-sw/cmssw/blob/master/DataFormats/Math/interface/deltaPhi.h to manage angle operations, I think we should avoid proliferating ways to do the same thing, especially by using not needed external dependencies

Reply:
It is already fixed, i.e. now we use the `M_PI` macro and common `reco::deltaPhi` function.